packages/actions/resources/views/components/action.blade.php

```diff
@@ -7,31 +7,34 @@
 @php
     $isDisabled = $action->isDisabled();
     $url = $action->getUrl();
+    $tooltip = $action->getTooltip();
+    $tooltipIsNotEmpty = str($tooltip)->isNotEmpty();
 @endphp
 
-<x-dynamic-component
-    :color="$action->getColor()"
-    :component="$dynamicComponent"
-    :disabled="$isDisabled"
-    :form="$action->getFormToSubmit()"
-    :form-id="$action->getFormId()"
-    :href="$isDisabled ? null : $url"
-    :icon="$icon ?? $action->getIcon()"
-    :icon-size="$action->getIconSize()"
-    :key-bindings="$action->getKeyBindings()"
-    :label-sr-only="$action->isLabelHidden()"
-    :tag="$url ? 'a' : 'button'"
-    :target="($url && $action->shouldOpenUrlInNewTab()) ? '_blank' : null"
-    :tooltip="$action->getTooltip()"
-    :type="$action->canSubmitForm() ? 'submit' : 'button'"
-    :wire:click="$action->getLivewireClickHandler()"
-    :wire:target="$action->getLivewireTarget()"
-    :x-on:click="$action->getAlpineClickHandler()"
-    :attributes="
-        \Filament\Support\prepare_inherited_attributes($attributes)
-            ->merge($action->getExtraAttributes(), escape: false)
-            ->class(['fi-ac-action'])
-    "
->
-    {{ $slot }}
-</x-dynamic-component>
+@if ($tooltipIsNotEmpty) <div x-data x-tooltip.raw="{{ $tooltip }}" class="flex"> @endif
+    <x-dynamic-component
+        :color="$action->getColor()"
+        :component="$dynamicComponent"
+        :disabled="$isDisabled"
+        :form="$action->getFormToSubmit()"
+        :form-id="$action->getFormId()"
+        :href="$isDisabled ? null : $url"
+        :icon="$icon ?? $action->getIcon()"
```

Comment:
What will be the scenario where $icon is null?

Reply:
checkout the video [here](https://github.com/filamentphp/filament/pull/13671#issuecomment-2248645952), where icon is null in all btns. or u mean something else?